### ISSUE TITLE
Enhancement to override forked-daapd controls

### ISSUE THREAD

**Issue by aleszczynskig** [ejurgensen/forked-daapd#1197]
Hello

would you consider providing an option to override the playback controls in forked-daapd under certain crircumstances? My use case here would be to provide a control interface to a device outputting to forked-daapd using a pipe. For example, if playing from a pipe, when clicking the next or previous buttons, instead of forked-daapd moving to the next or previous item in the queue, it would call or execute a script specified in the configuration file to control the source device. This would also enable hardware control of the source device which would be a really cool addition to the capability of forked-daapd.

This would be useful for example when using spocon for Spotify connect or integrating with any number of other source programs that may be using a pipe (of which I have a few).

**Comment by ejurgensen** [comment]
I think this would be fine, but it is not on the top of my list. Specifically with regard to Spotify Connect I would rather spend time on making it work out of the box.

One question would be on what condition(s) override the normal controls. I don't imagine you would always want it. An option would be to apply it only to pipe playback, i.e. make it conditional on the source data type.

**Comment by aleszczynskig** [comment]
Sure. I understand. A native Spotify connect capability would be awesome. 

In terms of conditions, yes. I guess the obvious use case would be for select pipe inputs. Maybe it would be possible to specify the list of pipe inputs which would override the controls and a script to call. Maybe the type of control could be passed to the script as a set of input parameters. Shairport-sync has a similar capability for volume and playback start/stop which might be a good reference, if needed.

**Comment by ejurgensen** [comment]
I've merged it into master now. Since the change is fairly safe I also included it in the 27.3 release that I just made. It will be a few days before I make an update to my RPi repo with the new version.

**Comment by aleszczynskig** [comment]
Great news. I wasn't expecting this for a while. Thank you

**Comment by aleszczynskig** [comment]
@ejurgensen - did you actually merge this functionality into 27.3? I can't see any reference to this capability. Is this referring to the pfls metadata support you added?

**Comment by ejurgensen** [comment]
Indeed, I mixed up the issues. It was the plfs support that was merged.
